src/sentry/api/serializers/models/rule.py

```diff
@@ -289,3 +302,192 @@ def serialize(self, obj, attrs, user, **kwargs) -> RuleSerializerResponse:
             d["disableDate"] = attrs["disable_date"]
 
         return d
+
+
+class WorkflowEngineRuleSerializer(Serializer):
+    def __init__(
+        self,
+        expand: list[str] | None = None,
+        prepare_component_fields: bool = False,
+        project_slug: str | None = None,
+    ):
+        super().__init__()
+        self.expand = expand or []
+        self.prepare_component_fields = prepare_component_fields
+        self.project_slug = project_slug
+
+    def get_attrs(self, item_list: Sequence[Workflow], user, **kwargs):
+        from sentry.workflow_engine.migration_helpers.rule_conditions import (
+            translate_to_rule_condition_filters,
+        )
+
+        # Bulk fetch environments
+        environments = Environment.objects.in_bulk(
+            [_f for _f in [i.environment_id for i in item_list] if _f]
+        )
+
+        # Bulk fetch users that created workflows
+        users = {
+            u.id: u
+            for u in user_service.get_many_by_id(
```

Comment:
```suggestion
            user.id: user
            for user in user_service.get_many_by_id(
```
it's a bit confusing with u and item being mixed in, better to just be explicit in naming - especially because it's already spanning multiple lines.